packages/shopping/src/controllers/user.controller.ts

```diff
@@ -260,4 +261,47 @@ export class UserController {
 
     return {token};
   }
+
+  @put('/users/password-reset', {
+    security: OPERATION_SECURITY_SPEC,
+    responses: {
+      '200': {
+        description: 'The updated user profile',
+        content: {
+          'application/json': {
+            schema: UserProfileSchema,
+          },
+        },
+      },
+    },
+  })
+  @authenticate('jwt')
+  async passwordReset(
+    @inject(SecurityBindings.USER)
+    currentUserProfile: UserProfile,
+    @requestBody(PasswordResetRequestBody) credentials: Credentials,
+  ): Promise<User> {
+    const {email, password} = credentials;
+    const {id} = currentUserProfile;
+
+    const user = await this.userRepository.findById(id);
+
+    if (!user) {
+      throw new HttpErrors.NotFound('User account not found');
+    }
+
+    if (email !== user?.email) {
+      throw new HttpErrors.Forbidden('Invalid email address');
+    }
+
+    validateCredentials(_.pick(credentials, ['email', 'password']));
+
+    const passwordHash = await this.passwordHasher.hashPassword(password);
+
+    await this.userRepository
+      .userCredentials(user.id)
+      .patch({password: passwordHash});
+
+    return user;
```

Comment:
Instead of returning the user, please generate a new token and return it, basically a re-login.

Reply:
Change included